velox/expression/CastExpr.cpp

```diff
@@ -515,6 +546,23 @@ VectorPtr CastExpr::applyDecimal(
   return castResult;
 }
 
+folly::Function<VectorPtr(
+    const SelectivityVector&,
+    const BaseVector&,
+    exec::EvalCtx&,
+    const TypePtr&)>
+getDecimalConverter(const bool isShortDecimal, const TypePtr& toType) {
+  if (toType->isDouble()) {
+    return isShortDecimal ? applyDecimalToDoubleCast<int64_t>
+                          : applyDecimalToDoubleCast<int128_t>;
+  } else if (toType->isInteger()) {
+    return isShortDecimal ? applyDecimalToIntCast<int64_t>
+                          : applyDecimalToIntCast<int128_t>;
+  }
+
+  return nullptr;
```

Comment:
This can be removed as this place is unreachable.

Reply:
We should be throwing an unsupported error here for the toType?

Reply:
@mbasmanova That place is reachable for types such as float, short etc?

Reply:
@majetideepak thought null access would do it, when it is called. but you are right, unsupported is much better exception.